fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -3647,7 +3657,8 @@ public CompletableFuture<Void> rebuildIndex(@Nonnull final Index index, @Nullabl
                     return null;
                 });
 
-        return context.instrument(FDBStoreTimer.Events.REBUILD_INDEX, future, startTime);
+        return CompletableFuture.allOf(context.instrument(reason.event, future, startTime),
+                context.instrument(FDBStoreTimer.Events.REBUILD_INDEX, future, startTime));
```

Comment:
It probably doesn't matter one way or the other, but it might be simper to have:
```
context.instrument(FDBStoreTimer.Events.REBUILD_INDEX, 
        context.instrument(reason.event, future, startTime),
        startTime);
```

Reply:
✅